interfaces/cython/cantera/onedim.py

```diff
@@ -545,147 +487,142 @@ def from_pandas(self, df, restore_boundaries=True, settings=None):
         requires a working pandas installation. The package 'pandas' can be
         installed using pip or conda.
         """
-        arr = SolutionArray(self.gas, extra=self._extra)
+        arr = SolutionArray(self.gas, extra=self.other())
         arr.from_pandas(df)
         self.from_solution_array(arr, restore_boundaries=restore_boundaries,
                                  settings=settings)
 
-    def write_hdf(self, filename, key=None, species='X',
-                  mode=None, complevel=None, quiet=True):
-        """
-        Write the solution vector to a HDF container file. Note that it is
-        possible to write multiple data entries to a single HDF container file.
-        Simulation settings are stored in tabular form as a separate HDF group
-        named ``settings``.
+    def write_hdf(self, filename, *args, group=None, species='X', mode='a',
+                  description=None, compression=None, compression_opts=None,
+                  quiet=True, **kwargs):
+        """
+        Write the solution vector to a HDF container file.
+
+        The `write_hdf` method preserves the stucture of a `FlameBase`-derived
+        object (i.e. `FreeFlame`, etc.). Each simulation is saved as a *group*,
```

Comment:
A minor style suggestion that @bryanwweber has gotten me started on is to avoid the use of these Latin abbreviations as a way of making things more accessible to non-fluent English speakers.
```suggestion
        object (such as `FreeFlame`). Each simulation is saved as a *group*,
```